bcipy/gui/BCInterface.py

```diff
@@ -108,25 +107,13 @@ def create_experiment(self) -> None:
 
         self.update_experiment_list()
 
-    def fast_scandir(self, directory_name: str, return_path: bool = True) -> List[str]:
-        """Fast Scan Directory.
-
-        directory_name: name of the directory to be scanned
-        return_path: whether or not to return the scanned directories as a relative path or name.
-            False will return the directory name only.
-        """
-        if return_path:
-            return [f.path for f in os.scandir(directory_name) if f.is_dir()]
-
-        return [f.name for f in os.scandir(directory_name) if f.is_dir()]
-
     def update_user_list(self) -> None:
         """Updates the user_input combo box with a list of user ids based on the
         data directory configured in the current parameters."""
 
         self.user_input.clear()
         self.user_input.addItem(BCInterface.default_text)
-        self.user_input.addItems(self.load_users())
+        self.user_input.addItems(load_users(self.parameters))
```

Comment:
Instead of passing all of the parameters, let's extract the data path in advance and pass it to your method!